src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
+    public void growIfArrayIsFull() {
+        if (size == elements.length) {
+            float multipler = 1.5f;
+            int newCapacity = Math.round(elements.length * multipler);
+            System.arraycopy(elements, 0, elements = new Object[newCapacity], 0, size);
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        growIfArrayIsFull();
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bounds: " + index
+                    + ", Size: " + size);
+        } else {
+            growIfArrayIsFull();
+            System.arraycopy(elements, index, elements, index + 1, Math.abs(index - size));
```

Comment:
Use System.arraycopy() correctly to shift elements to the right when inserting a new value. The fourth parameter 'index + 1' is correct, but the fifth parameter should be 'size - index' to move the correct number of elements.